iib/web/api_v1.py

```diff
@@ -185,6 +185,48 @@ def get_build_logs(request_id):
         return flask.Response(f.read(), mimetype='text/plain')
 
 
+@api_v1.route('/builds/<int:request_id>/related_bundle_images')
+def get_related_bundle_images(request_id):
+    """
+    Retrieve the related bundle images from the bundle CSV for a regenerate-bundle request.
+
+    :param int request_id: the request ID that was passed in through the URL.
+    :rtype: flask.Response
+    :raise NotFound: if the request is not found or there are no logs for the request
+    :raise Gone: if the logs for the build request have been removed due to expiration
+    """
+    request_related_bundles_dir = flask.current_app.config['IIB_REQUEST_RELATED_BUNDLES_DIR']
+    if not request_related_bundles_dir:
+        raise NotFound()
+
+    request = Request.query.get_or_404(request_id)
+    if request.type_name != RequestTypeMapping.pretty(3):
+        return flask.Response(
+            f'The request {request_id} is of type {request.type_name}. '
+            'This endpoint is only valid for requests of type regenerate-bundle.',
+            mimetype='application/json',
+        )
+
+    related_bundles_file_path = os.path.join(
+        request_related_bundles_dir, f'{request_id}_related_bundles.json'
+    )
+    if not os.path.exists(related_bundles_file_path):
+        expired = request.temporary_data_expiration < datetime.utcnow()
+        if expired:
+            raise Gone(f'The related_bundles for the build request {request_id} no longer exist')
```

Comment:
Let's be consistent in the wording that is exposed to the user. The API endpoint is `/related_bundle_images` but this error refers to it as `related_bundles`. Let's pick one and stick with it. My vote goes to the shorter form.